src/python/gudhi/datasets/generators/points/torus.py

```diff
@@ -0,0 +1,52 @@
+# This file is part of the Gudhi Library - https://gudhi.inria.fr/ - which is released under MIT.
+# See file LICENSE or go to https://gudhi.inria.fr/licensing/ for full license details.
+# Author(s):       Hind Montassif
+#
+# Copyright (C) 2021 Inria
+#
+# Modification(s):
+#   - YYYY/MM Author: Description of the modification
+
+import numpy as np
+import math
+
+
+def generate_random_points(num_points, dim):
+
+    # Generate random angles of size num_points*dim
+    alpha = 2*math.pi*np.random.rand(num_points*dim)
+
+    # Based on angles, construct points of size num_points*dim on a circle and reshape the result in a num_points*2*dim array
+    array_points = np.asarray([[np.cos(a), np.sin(a)] for a in alpha]).ravel().reshape(num_points, 2*dim)
+    
+    return array_points
+
+
+def generate_grid_points(num_points, dim):
```

Comment:
That formula is not really what comes to mind when I think of a grid. I was thinking more like, for dim=2
```python
k=3
alpha=np.linspace(0,2*np.pi,k,endpoint=False)
p2=np.column_stack([np.cos(alpha),np.sin(alpha)])
r=np.arange(k)
return p2[np.column_stack([np.repeat(r,k),np.tile(r,k)])].reshape(-1,4)
```
which gives

      [[ 1.       ,  0.       ,  1.       ,  0.       ],
       [ 1.       ,  0.       , -0.5      ,  0.8660254],
       [ 1.       ,  0.       , -0.5      , -0.8660254],
       [-0.5      ,  0.8660254,  1.       ,  0.       ],
       [-0.5      ,  0.8660254, -0.5      ,  0.8660254],
       [-0.5      ,  0.8660254, -0.5      , -0.8660254],
       [-0.5      , -0.8660254,  1.       ,  0.       ],
       [-0.5      , -0.8660254, -0.5      ,  0.8660254],
       [-0.5      , -0.8660254, -0.5      , -0.8660254]]


More in line with what you wrote, it is would also be interesting to sample points on a curve on the torus. Basically, instead of using 2 angles u and v, we use a single parameter t, and define u as 3*t and v as 5*t for good values of 3 and 5, and choose the range of t so that the curve is closed.

Reply:
If I understood correctly, you're taking the problem the other way around ? At the end, we would have an array of shape `num_points^dim, 2*dim`. But doesn't the user want a `(num_points, 2*dim) `array ? Or would he be aware of this since he is requesting a grid ? Or should the function take it into account by setting `k = (requested_num_points_by_user)^1/dim `?

Reply:
Er, the code isn't meant as final (it doesn't even handle dim!=2, and there are probably nicer ways than repeat/tile), yes, k can be `num_points**(1./dim)`.
I was mostly commenting on the rest of the function. As you can see in my sample output, the first 2 coordinates represent only 3 points, each repeated 3 times. Same for the last 2 coordinates, but repeated differently, so we get the equivalent of a grid.
Your version looks very similar to the random case, and I don't see in what sense it represents a grid.

Reply:
Ok, yes I was more talking about the general idea of the code that the code itself.

Reply:
A consistant naming of the arguments also becomes important. If for the sphere we use n_samples for the number of points and dim for the ambient dimension, here we shouldn't use dim for the dimension of the torus, dim should also be the ambient dimension. Or we should rename dim to ambient_dim for the sphere, or... And if we wanted, we could instead of n_samples take an argument, say grid_size, which says how many values we consider on each axis of the grid (k in my code), and it would even be possible to accept a tuple for this argument if we want a different step size on different axes, but that complicates things.

Reply:
* we kept `dim` for the torus and `dim` has been renamed `ambient_dim` for the sphere.
* @mglisse is it ok for you to keep `n_samples` or would you prefer a `grid_size` argument ?